grails-app/controllers/au/org/ala/ecodata/ActivityController.groovy

```diff
@@ -51,6 +51,23 @@ class ActivityController {
         }
     }
 
+    /**
+     * Deletes all activities associated with project activityId.
+     *
+     * @param id project activity id
+     */
+    @RequireApiKey
+    def deleteByProjectActivity(String id) {
+        boolean destroy = params.destroy == null ? false : params.destroy.toBoolean()
+        if (activityService.deleteByProjectActivity(id, destroy).status == 'ok') {
```

Comment:
deleteByProjectActivity returns 3 possible statuses: ok, error, not found.

This code is treating 'error' and 'not found' the same way: returning a 404. 

A status of 'error' should probably return a 500.